Examples/Tests/secondary_ion_emission/inputs_test_rz_secondary_ion_emission_picmi.py

```diff
@@ -0,0 +1,255 @@
+    upper_boundary_conditions_particles=["absorbing", "reflecting"],
+)
+
+solver = picmi.ElectrostaticSolver(
+    grid=grid, method="Multigrid", warpx_absolute_tolerance=1e-7
+)
+
+embedded_boundary = picmi.EmbeddedBoundary(
+    implicit_function="-(x**2+y**2+z**2-radius**2)", radius=0.2
+)
+
+##########################
+# physics components
+##########################
+
+i_dist = picmi.ParticleListDistribution(
+    x=0.0, y=0.0, z=-0.25, ux=0.1e6, uy=0.0, uz=1.50e6, weight=1
+)
+electrons = picmi.Species(
+    particle_type="electron",  # Specify the particle type
+    name="electrons",  # Name of the species
+)
+
+ions = picmi.Species(
+    name="ions",
+    mass=proton_mass,
+    charge=e,
+    initial_distribution=i_dist,
+    warpx_save_particles_at_eb=1,
+)
+
+##########################
+# diagnostics
+##########################
+
+field_diag = picmi.FieldDiagnostic(
+    name="diag1",
+    grid=grid,
+    period=diagnostic_interval,
+    data_list=["Er", "Ez", "phi", "rho"],  # , "rho_electrons"],
+    warpx_format="openpmd",
+)
+
+part_diag = picmi.ParticleDiagnostic(
+    name="diag1",
+    period=diagnostic_interval,
+    species=[ions, electrons],
+    warpx_format="openpmd",
+)
+
+##########################
+# simulation setup
+##########################
+# num_procs = [1, 1]
+# warpx_numprocs=num_procs,
+sim = picmi.Simulation(
+    solver=solver,
+    time_step_size=dt,
+    max_steps=max_steps,
+    warpx_embedded_boundary=embedded_boundary,
+    warpx_amrex_the_arena_is_managed=1,
+)
+
+sim.add_species(
+    electrons,
+    layout=picmi.GriddedLayout(n_macroparticle_per_cell=[0, 0, 0], grid=grid),
+)
+
+sim.add_species(
+    ions,
+    layout=picmi.GriddedLayout(n_macroparticle_per_cell=[10, 1, 1], grid=grid),
+)
+
+sim.add_diagnostic(part_diag)
+sim.add_diagnostic(field_diag)
+
+sim.initialize_inputs()
+sim.initialize_warpx()
+
+##########################
+# python particle data access
+##########################
+
+
+def concat(list_of_arrays):
+    if len(list_of_arrays) == 0:
+        # Return a 1d array of size 0
+        return np.empty(0)
+    else:
+        return np.concatenate(list_of_arrays)
+
+
+def sigma_nascap(energy_kEv, delta_H, E_HMax):
+    """
+    Compute sigma_nascap for each element in the energy array using a loop.
+
+    Parameters:
+    - energy: ndarray or list, energy values in KeV
+    - delta_H: float, parameter for the formula
+    - E_HMax: float, parameter for the formula in KeV
+
+    Returns:
+    - numpy array, computed probability sigma_nascap
+    """
+    sigma_nascap = np.array([])
+    # Loop through each energy value
+    for energy in energy_kEv:
+        if energy > 0.0:
+            sigma = (
+                delta_H
+                * (E_HMax + 1.0)
+                / (E_HMax * 1.0 + energy)
+                * np.sqrt(energy / 1.0)
+            )
+        else:
+            sigma = 0.0
+        sigma_nascap = np.append(sigma_nascap, sigma)
```

Comment:
Could you try to see if this one-line list comprehension works and is equivalent to your implementation?
```suggestion
    sigma = np.array([(delta_H * (E_HMax + 1.0) / (E_HMax + energy) * np.sqrt(energy) if energy > 0.0 else 0.0) for energy in energy_kEv])
 ```

Reply:
The pre-commit hook enforces the repository’s style policy and divides it into multiple lines anyway, so I suggest leaving the code as-is instead of bypassing it.